server/game/cards/01_SOR/units/BobaFettDisintegrator.ts

```diff
@@ -0,0 +1,39 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { CardsPlayedThisPhaseWatcher } from '../../../stateWatchers/CardsPlayedThisPhaseWatcher';
+import { StateWatcherRegistrar } from '../../../core/stateWatcher/StateWatcherRegistrar';
+
+export default class BobaFettDisintegrator extends NonLeaderUnitCard {
+    // initiate watcher record
+    private cardsPlayedThisPhaseWatcher: CardsPlayedThisPhaseWatcher;
+
+    protected override getImplementationId() {
+        return {
+            id: '4156799805',
+            internalName: 'boba-fett#disintegrator'
+        };
+    }
+
+    // setup watcher
+    protected override setupStateWatchers(registrar: StateWatcherRegistrar) {
+        this.cardsPlayedThisPhaseWatcher = AbilityHelper.stateWatchers.cardsPlayedThisPhase(registrar, this);
```

Comment:
So as we discussed in the chat, there is a difference between "played" and "enters play", which is mostly important for leaders (leaders enter play but are not played).

The Boba effect is supposed to work the same for both leaders and units - i.e., if they entered play this phase (either by being deployed or being played from hand / smuggled), the Boba effect does not work on them. If they entered play the previous round, it does work.

So the first thing is that the `CardsPlayedThisPhaseWatcher` will not work for this card because it is only looking for played cards, not cards entering play. You will need to copy CardsPlayedThisPhaseWatcher.ts and create a new, slightly modified CardsEnteredPlayThisPhaseWatcher.ts that looks for `onUnitEntersPlay` instead of `onCardPlayed` in its `setupWatcher()` method.

The other issue is that currently there is a bug where the `onUnitEntersPlay` event is not firing correctly for leaders at all. Val is working on this in his PR. If he gets it merged soon enough then we can just merge it into yours, otherwise we can just comment out the leader tests for now (with a TODO comment explaining why) and do a new PR to update it once Val's PR is in.

Reply:
I added the CardsEnteredPlayedThisPhaseWatcher and updated the BobaFett implementation with it.